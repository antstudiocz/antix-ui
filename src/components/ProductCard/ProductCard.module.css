/* Product Card specific variables */
:root {
  --product-card-width: 269px;
  --product-card-height: 434px;
  --product-card-padding: 8px 8px 32px;
  --product-card-radius: var(--border-radius-xl);
  --product-card-image-height: 180px;
  --product-card-image-radius: var(--border-radius-lg);
  --product-card-shadow: 0px 1px 3px 0px var(--shadow-color-light), 0px 1px 2px 0px var(--input-shadow-color);
  
  /* Font sizes for product card */
  --font-product-title-size: 18px;
  --font-product-title-line-height: 1.2;
  --font-product-price-size: var(--font-button1-size);
  --font-product-price-line-height: var(--font-button1-line-height);
  --font-product-badge-size: var(--font-label-medium-size);
  --font-product-badge-line-height: var(--font-label-medium-line-height);

  /* Product card title */
  --product-card-title-line-clamp: 1; 
  --product-card-title-min-height: calc(var(--font-product-title-size) * var(--product-card-title-line-clamp) * var(--font-product-title-line-height));
}

.card {
  width: var(--product-card-width);
  max-height: var(--product-card-height);
  padding: var(--product-card-padding);
  position: relative;
  background: var(--color-neutral-00);
  border-radius: var(--product-card-radius);
  box-shadow: var(--product-card-shadow);
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 12px;
  overflow: hidden;
}

.image {
  align-self: stretch;
  height: var(--product-card-image-height);
  border-radius: var(--product-card-image-radius);
  object-fit: cover;
  flex-shrink: 0;
}

.title {
  align-self: stretch;
  text-align: center;
  color: var(--color-secondary-800);
  font-family: var(--font-secondary);
  font-size: var(--font-product-title-size);
  font-weight: 600;
  line-height: var(--font-product-title-line-height);
  margin: 0;
  min-height: var(--product-card-title-min-height);
  display: -webkit-box;
  -webkit-line-clamp: var(--product-card-title-line-clamp);
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.priceContainer {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  gap: 4px;
  flex-shrink: 0;
  width: 100%;
}

.priceOriginal {
  align-self: stretch;
  text-align: center;
  color: var(--color-conversion-500);
  font-family: var(--font-secondary);
  font-size: var(--font-button2-size);
  font-weight: 600;
  line-height: var(--font-button2-line-height);
  text-decoration: line-through;
}

.priceCurrent {
  align-self: stretch;
  text-align: center;
  color: var(--color-secondary-800);
  font-family: var(--font-secondary);
  font-size: var(--font-product-price-size);
  font-weight: 700;
  line-height: var(--font-product-price-line-height);
}

.bottomContainer {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  gap: 24px;
  width: 100%;
  margin-top: auto;
}

.availabilityContainer {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}

.availabilityItem {
  padding: 4px;
  display: inline-flex;
  justify-content: flex-start;
  align-items: center;
  gap: 4px;
  overflow: hidden;
}

.availabilityItem span {
  color: var(--color-success);
  font-family: var(--font-secondary);
  font-size: var(--font-button2-size);
  font-weight: 600;
  line-height: 1;
}

.availabilityItem svg {
  color: var(--color-success);
}

.badgesWrapper {
  position: absolute;
  left: 20px;
  top: 20px;
  display: flex;
  flex-direction: column;
  gap: 6px;
  z-index: 1;
}

.badge {
  padding: 4px 6px;
  background: #000000;
  color: #FFFFFF;
  border-radius: var(--border-radius-sm);
  display: inline-flex;
  justify-content: center;
  align-items: center;
  width: fit-content;
  gap: 10px;
  overflow: hidden;
  font-family: var(--font-secondary);
  font-size: var(--font-product-badge-size);
  font-weight: 600;
  line-height: var(--font-product-badge-line-height);
}
