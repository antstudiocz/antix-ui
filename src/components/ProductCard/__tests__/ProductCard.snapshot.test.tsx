import React from 'react';
import { render } from '@testing-library/react';
import { ProductCard } from '../ProductCard';
import '../../../translations/i18n';

describe('ProductCard Snapshots', () => {
  const defaultProps = {
    imageUrl: 'https://example.com/product-image.jpg',
    title: 'Test Product',
    currentPrice: '299.00 Kč / pc',
    onAddToCart: jest.fn(),
  };

  it('matches snapshot with minimal props', () => {
    const { container } = render(<ProductCard {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with all props', () => {
    const badges = [
      { text: 'Novinka' },
      { text: 'Akce', backgroundColor: '#FF0000', textColor: '#FFFFFF' }
    ];

    const { container } = render(
      <ProductCard 
        {...defaultProps}
        originalPrice="399.00 Kč / pc"
        badges={badges}
        deliveryStatus="Doručíme dnes"
        storeAvailability="Ihned na prodejně"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with original price only', () => {
    const { container } = render(
      <ProductCard 
        {...defaultProps}
        originalPrice="399.00 Kč / pc"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with badges only', () => {
    const badges = [
      { text: 'Novinka' },
      { text: 'Akce', backgroundColor: '#FF0000', textColor: '#FFFFFF' }
    ];

    const { container } = render(
      <ProductCard 
        {...defaultProps}
        badges={badges}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with delivery status only', () => {
    const { container } = render(
      <ProductCard 
        {...defaultProps}
        deliveryStatus="Doručíme dnes"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with store availability only', () => {
    const { container } = render(
      <ProductCard 
        {...defaultProps}
        storeAvailability="Ihned na prodejně"
      />
    );
    expect(container).toMatchSnapshot();
  });
}); 