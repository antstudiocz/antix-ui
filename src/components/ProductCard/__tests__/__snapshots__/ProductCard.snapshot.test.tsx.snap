// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Snapshots matches snapshot with all props 1`] = `
<div>
  <div
    class="w-[var(--product-card-width)] max-h-[var(--product-card-height)] p-2 pb-8 relative bg-neutral-00 rounded-xl shadow-sm flex flex-col items-center gap-3 overflow-hidden"
  >
    <img
      alt="Test Product"
      class="self-stretch h-[var(--product-card-image-height)] rounded-lg object-cover flex-shrink-0"
      src="https://example.com/product-image.jpg"
    />
    <h4
      class="self-stretch text-center text-secondary-800 font-secondary text-product-title font-semibold m-0 min-h-[var(--product-card-title-min-height)] line-clamp-[var(--product-card-title-line-clamp)] overflow-hidden"
    >
      Test Product
    </h4>
    <div
      class="flex flex-col justify-start items-center gap-1 flex-shrink-0 w-full"
    >
      <div
        class="self-stretch text-center text-conversion-500 font-secondary text-product-price-original font-semibold line-through"
      >
        <span>
          399.00 Kč / pc
        </span>
      </div>
      <div
        class="self-stretch text-center text-secondary-800 font-secondary text-product-price-current"
      >
        299.00 Kč / pc
      </div>
    </div>
    <div
      class="flex flex-col justify-start items-center gap-6 w-full mt-auto"
    >
      <div
        class="flex flex-col justify-start items-center"
      >
        <div
          class="p-1 inline-flex justify-start items-center gap-1 overflow-hidden text-success "
        >
          <svg
            class="stroke-current"
            fill="none"
            height="12"
            viewBox="0 0 16 12"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.33325 9.99996V1.99996C9.33325 1.64634 9.19278 1.3072 8.94273 1.05715C8.69268 0.807102 8.35354 0.666626 7.99992 0.666626H2.66659C2.31296 0.666626 1.97382 0.807102 1.72378 1.05715C1.47373 1.3072 1.33325 1.64634 1.33325 1.99996V9.33329C1.33325 9.5101 1.40349 9.67967 1.52851 9.8047C1.65354 9.92972 1.82311 9.99996 1.99992 9.99996H3.33325M3.33325 9.99996C3.33325 10.7363 3.93021 11.3333 4.66659 11.3333C5.40297 11.3333 5.99992 10.7363 5.99992 9.99996M3.33325 9.99996C3.33325 9.26358 3.93021 8.66663 4.66659 8.66663C5.40297 8.66663 5.99992 9.26358 5.99992 9.99996M9.99992 9.99996H5.99992M9.99992 9.99996C9.99992 10.7363 10.5969 11.3333 11.3333 11.3333C12.0696 11.3333 12.6666 10.7363 12.6666 9.99996M9.99992 9.99996C9.99992 9.26358 10.5969 8.66663 11.3333 8.66663C12.0696 8.66663 12.6666 9.26358 12.6666 9.99996M12.6666 9.99996H13.9999C14.1767 9.99996 14.3463 9.92972 14.4713 9.8047C14.5963 9.67967 14.6666 9.5101 14.6666 9.33329V6.89996C14.6663 6.74867 14.6146 6.60197 14.5199 6.48396L12.1999 3.58396C12.1376 3.50588 12.0585 3.44281 11.9685 3.39942C11.8784 3.35604 11.7798 3.33343 11.6799 3.33329H9.33325"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
          <span
            class="font-secondary text-button-md font-semibold"
          >
            Doručíme dnes
          </span>
        </div>
        <div
          class="p-1 inline-flex justify-start items-center gap-1 overflow-hidden text-success"
        >
          <svg
            class="stroke-current"
            fill="none"
            height="9"
            viewBox="0 0 12 9"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.3334 1L4.00008 8.33333L0.666748 5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
          <span
            class="font-secondary text-button-md font-semibold"
          >
            Ihned na prodejně
          </span>
        </div>
      </div>
      <button
        aria-label="Add to cart"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-xl font-secondary font-bold transition-all duration-200 text-center relative box-border h-fit disabled:opacity-60 disabled:cursor-not-allowed cursor-pointer px-6 py-3.5 text-conversion-500 flex-row bg-conversion-500 *:text-neutral-00 hover:bg-conversion-900 gap-2.5 text-button-lg"
      >
        <span
          class="icon inline-flex items-center justify-center text-current"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 19 20"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.20825 1.70837H2.87492L5.09159 12.0584C5.1729 12.4374 5.38381 12.7763 5.68801 13.0166C5.99221 13.2569 6.37067 13.3837 6.75825 13.375H14.9083C15.2876 13.3744 15.6553 13.2444 15.9508 13.0066C16.2462 12.7687 16.4517 12.4371 16.5333 12.0667L17.9083 5.87504H3.76659M6.99992 17.5C6.99992 17.9603 6.62682 18.3334 6.16659 18.3334C5.70635 18.3334 5.33325 17.9603 5.33325 17.5C5.33325 17.0398 5.70635 16.6667 6.16659 16.6667C6.62682 16.6667 6.99992 17.0398 6.99992 17.5ZM16.1666 17.5C16.1666 17.9603 15.7935 18.3334 15.3333 18.3334C14.873 18.3334 14.4999 17.9603 14.4999 17.5C14.4999 17.0398 14.873 16.6667 15.3333 16.6667C15.7935 16.6667 16.1666 17.0398 16.1666 17.5Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
          </svg>
        </span>
        <span
          class="inline-block whitespace-nowrap overflow-visible text-current"
        >
          Add to cart
        </span>
      </button>
    </div>
    <div
      class="absolute left-5 top-5 flex flex-col gap-1.5 z-10"
    >
      <span
        class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-primary text-primary-foreground [a&]:hover:bg-primary/90 font-secondary !text-product-badge font-semibold"
        data-slot="badge"
        style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
      >
        Novinka
      </span>
      <span
        class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-primary text-primary-foreground [a&]:hover:bg-primary/90 font-secondary !text-product-badge font-semibold"
        data-slot="badge"
        style="background-color: rgb(255, 0, 0); color: rgb(255, 255, 255);"
      >
        Akce
      </span>
    </div>
  </div>
</div>
`;

exports[`ProductCard Snapshots matches snapshot with badges only 1`] = `
<div>
  <div
    class="w-[var(--product-card-width)] max-h-[var(--product-card-height)] p-2 pb-8 relative bg-neutral-00 rounded-xl shadow-sm flex flex-col items-center gap-3 overflow-hidden"
  >
    <img
      alt="Test Product"
      class="self-stretch h-[var(--product-card-image-height)] rounded-lg object-cover flex-shrink-0"
      src="https://example.com/product-image.jpg"
    />
    <h4
      class="self-stretch text-center text-secondary-800 font-secondary text-product-title font-semibold m-0 min-h-[var(--product-card-title-min-height)] line-clamp-[var(--product-card-title-line-clamp)] overflow-hidden"
    >
      Test Product
    </h4>
    <div
      class="flex flex-col justify-start items-center gap-1 flex-shrink-0 w-full"
    >
      <div
        class="self-stretch text-center text-secondary-800 font-secondary text-product-price-current"
      >
        299.00 Kč / pc
      </div>
    </div>
    <div
      class="flex flex-col justify-start items-center gap-6 w-full mt-auto"
    >
      <div
        class="flex flex-col justify-start items-center"
      />
      <button
        aria-label="Add to cart"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-xl font-secondary font-bold transition-all duration-200 text-center relative box-border h-fit disabled:opacity-60 disabled:cursor-not-allowed cursor-pointer px-6 py-3.5 text-conversion-500 flex-row bg-conversion-500 *:text-neutral-00 hover:bg-conversion-900 gap-2.5 text-button-lg"
      >
        <span
          class="icon inline-flex items-center justify-center text-current"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 19 20"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.20825 1.70837H2.87492L5.09159 12.0584C5.1729 12.4374 5.38381 12.7763 5.68801 13.0166C5.99221 13.2569 6.37067 13.3837 6.75825 13.375H14.9083C15.2876 13.3744 15.6553 13.2444 15.9508 13.0066C16.2462 12.7687 16.4517 12.4371 16.5333 12.0667L17.9083 5.87504H3.76659M6.99992 17.5C6.99992 17.9603 6.62682 18.3334 6.16659 18.3334C5.70635 18.3334 5.33325 17.9603 5.33325 17.5C5.33325 17.0398 5.70635 16.6667 6.16659 16.6667C6.62682 16.6667 6.99992 17.0398 6.99992 17.5ZM16.1666 17.5C16.1666 17.9603 15.7935 18.3334 15.3333 18.3334C14.873 18.3334 14.4999 17.9603 14.4999 17.5C14.4999 17.0398 14.873 16.6667 15.3333 16.6667C15.7935 16.6667 16.1666 17.0398 16.1666 17.5Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
          </svg>
        </span>
        <span
          class="inline-block whitespace-nowrap overflow-visible text-current"
        >
          Add to cart
        </span>
      </button>
    </div>
    <div
      class="absolute left-5 top-5 flex flex-col gap-1.5 z-10"
    >
      <span
        class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-primary text-primary-foreground [a&]:hover:bg-primary/90 font-secondary !text-product-badge font-semibold"
        data-slot="badge"
        style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
      >
        Novinka
      </span>
      <span
        class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-primary text-primary-foreground [a&]:hover:bg-primary/90 font-secondary !text-product-badge font-semibold"
        data-slot="badge"
        style="background-color: rgb(255, 0, 0); color: rgb(255, 255, 255);"
      >
        Akce
      </span>
    </div>
  </div>
</div>
`;

exports[`ProductCard Snapshots matches snapshot with delivery status only 1`] = `
<div>
  <div
    class="w-[var(--product-card-width)] max-h-[var(--product-card-height)] p-2 pb-8 relative bg-neutral-00 rounded-xl shadow-sm flex flex-col items-center gap-3 overflow-hidden"
  >
    <img
      alt="Test Product"
      class="self-stretch h-[var(--product-card-image-height)] rounded-lg object-cover flex-shrink-0"
      src="https://example.com/product-image.jpg"
    />
    <h4
      class="self-stretch text-center text-secondary-800 font-secondary text-product-title font-semibold m-0 min-h-[var(--product-card-title-min-height)] line-clamp-[var(--product-card-title-line-clamp)] overflow-hidden"
    >
      Test Product
    </h4>
    <div
      class="flex flex-col justify-start items-center gap-1 flex-shrink-0 w-full"
    >
      <div
        class="self-stretch text-center text-secondary-800 font-secondary text-product-price-current"
      >
        299.00 Kč / pc
      </div>
    </div>
    <div
      class="flex flex-col justify-start items-center gap-6 w-full mt-auto"
    >
      <div
        class="flex flex-col justify-start items-center"
      >
        <div
          class="p-1 inline-flex justify-start items-center gap-1 overflow-hidden text-success "
        >
          <svg
            class="stroke-current"
            fill="none"
            height="12"
            viewBox="0 0 16 12"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.33325 9.99996V1.99996C9.33325 1.64634 9.19278 1.3072 8.94273 1.05715C8.69268 0.807102 8.35354 0.666626 7.99992 0.666626H2.66659C2.31296 0.666626 1.97382 0.807102 1.72378 1.05715C1.47373 1.3072 1.33325 1.64634 1.33325 1.99996V9.33329C1.33325 9.5101 1.40349 9.67967 1.52851 9.8047C1.65354 9.92972 1.82311 9.99996 1.99992 9.99996H3.33325M3.33325 9.99996C3.33325 10.7363 3.93021 11.3333 4.66659 11.3333C5.40297 11.3333 5.99992 10.7363 5.99992 9.99996M3.33325 9.99996C3.33325 9.26358 3.93021 8.66663 4.66659 8.66663C5.40297 8.66663 5.99992 9.26358 5.99992 9.99996M9.99992 9.99996H5.99992M9.99992 9.99996C9.99992 10.7363 10.5969 11.3333 11.3333 11.3333C12.0696 11.3333 12.6666 10.7363 12.6666 9.99996M9.99992 9.99996C9.99992 9.26358 10.5969 8.66663 11.3333 8.66663C12.0696 8.66663 12.6666 9.26358 12.6666 9.99996M12.6666 9.99996H13.9999C14.1767 9.99996 14.3463 9.92972 14.4713 9.8047C14.5963 9.67967 14.6666 9.5101 14.6666 9.33329V6.89996C14.6663 6.74867 14.6146 6.60197 14.5199 6.48396L12.1999 3.58396C12.1376 3.50588 12.0585 3.44281 11.9685 3.39942C11.8784 3.35604 11.7798 3.33343 11.6799 3.33329H9.33325"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
          <span
            class="font-secondary text-button-md font-semibold"
          >
            Doručíme dnes
          </span>
        </div>
      </div>
      <button
        aria-label="Add to cart"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-xl font-secondary font-bold transition-all duration-200 text-center relative box-border h-fit disabled:opacity-60 disabled:cursor-not-allowed cursor-pointer px-6 py-3.5 text-conversion-500 flex-row bg-conversion-500 *:text-neutral-00 hover:bg-conversion-900 gap-2.5 text-button-lg"
      >
        <span
          class="icon inline-flex items-center justify-center text-current"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 19 20"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.20825 1.70837H2.87492L5.09159 12.0584C5.1729 12.4374 5.38381 12.7763 5.68801 13.0166C5.99221 13.2569 6.37067 13.3837 6.75825 13.375H14.9083C15.2876 13.3744 15.6553 13.2444 15.9508 13.0066C16.2462 12.7687 16.4517 12.4371 16.5333 12.0667L17.9083 5.87504H3.76659M6.99992 17.5C6.99992 17.9603 6.62682 18.3334 6.16659 18.3334C5.70635 18.3334 5.33325 17.9603 5.33325 17.5C5.33325 17.0398 5.70635 16.6667 6.16659 16.6667C6.62682 16.6667 6.99992 17.0398 6.99992 17.5ZM16.1666 17.5C16.1666 17.9603 15.7935 18.3334 15.3333 18.3334C14.873 18.3334 14.4999 17.9603 14.4999 17.5C14.4999 17.0398 14.873 16.6667 15.3333 16.6667C15.7935 16.6667 16.1666 17.0398 16.1666 17.5Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
          </svg>
        </span>
        <span
          class="inline-block whitespace-nowrap overflow-visible text-current"
        >
          Add to cart
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`ProductCard Snapshots matches snapshot with minimal props 1`] = `
<div>
  <div
    class="w-[var(--product-card-width)] max-h-[var(--product-card-height)] p-2 pb-8 relative bg-neutral-00 rounded-xl shadow-sm flex flex-col items-center gap-3 overflow-hidden"
  >
    <img
      alt="Test Product"
      class="self-stretch h-[var(--product-card-image-height)] rounded-lg object-cover flex-shrink-0"
      src="https://example.com/product-image.jpg"
    />
    <h4
      class="self-stretch text-center text-secondary-800 font-secondary text-product-title font-semibold m-0 min-h-[var(--product-card-title-min-height)] line-clamp-[var(--product-card-title-line-clamp)] overflow-hidden"
    >
      Test Product
    </h4>
    <div
      class="flex flex-col justify-start items-center gap-1 flex-shrink-0 w-full"
    >
      <div
        class="self-stretch text-center text-secondary-800 font-secondary text-product-price-current"
      >
        299.00 Kč / pc
      </div>
    </div>
    <div
      class="flex flex-col justify-start items-center gap-6 w-full mt-auto"
    >
      <div
        class="flex flex-col justify-start items-center"
      />
      <button
        aria-label="Add to cart"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-xl font-secondary font-bold transition-all duration-200 text-center relative box-border h-fit disabled:opacity-60 disabled:cursor-not-allowed cursor-pointer px-6 py-3.5 text-conversion-500 flex-row bg-conversion-500 *:text-neutral-00 hover:bg-conversion-900 gap-2.5 text-button-lg"
      >
        <span
          class="icon inline-flex items-center justify-center text-current"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 19 20"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.20825 1.70837H2.87492L5.09159 12.0584C5.1729 12.4374 5.38381 12.7763 5.68801 13.0166C5.99221 13.2569 6.37067 13.3837 6.75825 13.375H14.9083C15.2876 13.3744 15.6553 13.2444 15.9508 13.0066C16.2462 12.7687 16.4517 12.4371 16.5333 12.0667L17.9083 5.87504H3.76659M6.99992 17.5C6.99992 17.9603 6.62682 18.3334 6.16659 18.3334C5.70635 18.3334 5.33325 17.9603 5.33325 17.5C5.33325 17.0398 5.70635 16.6667 6.16659 16.6667C6.62682 16.6667 6.99992 17.0398 6.99992 17.5ZM16.1666 17.5C16.1666 17.9603 15.7935 18.3334 15.3333 18.3334C14.873 18.3334 14.4999 17.9603 14.4999 17.5C14.4999 17.0398 14.873 16.6667 15.3333 16.6667C15.7935 16.6667 16.1666 17.0398 16.1666 17.5Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
          </svg>
        </span>
        <span
          class="inline-block whitespace-nowrap overflow-visible text-current"
        >
          Add to cart
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`ProductCard Snapshots matches snapshot with original price only 1`] = `
<div>
  <div
    class="w-[var(--product-card-width)] max-h-[var(--product-card-height)] p-2 pb-8 relative bg-neutral-00 rounded-xl shadow-sm flex flex-col items-center gap-3 overflow-hidden"
  >
    <img
      alt="Test Product"
      class="self-stretch h-[var(--product-card-image-height)] rounded-lg object-cover flex-shrink-0"
      src="https://example.com/product-image.jpg"
    />
    <h4
      class="self-stretch text-center text-secondary-800 font-secondary text-product-title font-semibold m-0 min-h-[var(--product-card-title-min-height)] line-clamp-[var(--product-card-title-line-clamp)] overflow-hidden"
    >
      Test Product
    </h4>
    <div
      class="flex flex-col justify-start items-center gap-1 flex-shrink-0 w-full"
    >
      <div
        class="self-stretch text-center text-conversion-500 font-secondary text-product-price-original font-semibold line-through"
      >
        <span>
          399.00 Kč / pc
        </span>
      </div>
      <div
        class="self-stretch text-center text-secondary-800 font-secondary text-product-price-current"
      >
        299.00 Kč / pc
      </div>
    </div>
    <div
      class="flex flex-col justify-start items-center gap-6 w-full mt-auto"
    >
      <div
        class="flex flex-col justify-start items-center"
      />
      <button
        aria-label="Add to cart"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-xl font-secondary font-bold transition-all duration-200 text-center relative box-border h-fit disabled:opacity-60 disabled:cursor-not-allowed cursor-pointer px-6 py-3.5 text-conversion-500 flex-row bg-conversion-500 *:text-neutral-00 hover:bg-conversion-900 gap-2.5 text-button-lg"
      >
        <span
          class="icon inline-flex items-center justify-center text-current"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 19 20"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.20825 1.70837H2.87492L5.09159 12.0584C5.1729 12.4374 5.38381 12.7763 5.68801 13.0166C5.99221 13.2569 6.37067 13.3837 6.75825 13.375H14.9083C15.2876 13.3744 15.6553 13.2444 15.9508 13.0066C16.2462 12.7687 16.4517 12.4371 16.5333 12.0667L17.9083 5.87504H3.76659M6.99992 17.5C6.99992 17.9603 6.62682 18.3334 6.16659 18.3334C5.70635 18.3334 5.33325 17.9603 5.33325 17.5C5.33325 17.0398 5.70635 16.6667 6.16659 16.6667C6.62682 16.6667 6.99992 17.0398 6.99992 17.5ZM16.1666 17.5C16.1666 17.9603 15.7935 18.3334 15.3333 18.3334C14.873 18.3334 14.4999 17.9603 14.4999 17.5C14.4999 17.0398 14.873 16.6667 15.3333 16.6667C15.7935 16.6667 16.1666 17.0398 16.1666 17.5Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
          </svg>
        </span>
        <span
          class="inline-block whitespace-nowrap overflow-visible text-current"
        >
          Add to cart
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`ProductCard Snapshots matches snapshot with store availability only 1`] = `
<div>
  <div
    class="w-[var(--product-card-width)] max-h-[var(--product-card-height)] p-2 pb-8 relative bg-neutral-00 rounded-xl shadow-sm flex flex-col items-center gap-3 overflow-hidden"
  >
    <img
      alt="Test Product"
      class="self-stretch h-[var(--product-card-image-height)] rounded-lg object-cover flex-shrink-0"
      src="https://example.com/product-image.jpg"
    />
    <h4
      class="self-stretch text-center text-secondary-800 font-secondary text-product-title font-semibold m-0 min-h-[var(--product-card-title-min-height)] line-clamp-[var(--product-card-title-line-clamp)] overflow-hidden"
    >
      Test Product
    </h4>
    <div
      class="flex flex-col justify-start items-center gap-1 flex-shrink-0 w-full"
    >
      <div
        class="self-stretch text-center text-secondary-800 font-secondary text-product-price-current"
      >
        299.00 Kč / pc
      </div>
    </div>
    <div
      class="flex flex-col justify-start items-center gap-6 w-full mt-auto"
    >
      <div
        class="flex flex-col justify-start items-center"
      >
        <div
          class="p-1 inline-flex justify-start items-center gap-1 overflow-hidden text-success"
        >
          <svg
            class="stroke-current"
            fill="none"
            height="9"
            viewBox="0 0 12 9"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.3334 1L4.00008 8.33333L0.666748 5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.33333"
            />
          </svg>
          <span
            class="font-secondary text-button-md font-semibold"
          >
            Ihned na prodejně
          </span>
        </div>
      </div>
      <button
        aria-label="Add to cart"
        class="inline-flex items-center justify-center whitespace-nowrap rounded-xl font-secondary font-bold transition-all duration-200 text-center relative box-border h-fit disabled:opacity-60 disabled:cursor-not-allowed cursor-pointer px-6 py-3.5 text-conversion-500 flex-row bg-conversion-500 *:text-neutral-00 hover:bg-conversion-900 gap-2.5 text-button-lg"
      >
        <span
          class="icon inline-flex items-center justify-center text-current"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 19 20"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.20825 1.70837H2.87492L5.09159 12.0584C5.1729 12.4374 5.38381 12.7763 5.68801 13.0166C5.99221 13.2569 6.37067 13.3837 6.75825 13.375H14.9083C15.2876 13.3744 15.6553 13.2444 15.9508 13.0066C16.2462 12.7687 16.4517 12.4371 16.5333 12.0667L17.9083 5.87504H3.76659M6.99992 17.5C6.99992 17.9603 6.62682 18.3334 6.16659 18.3334C5.70635 18.3334 5.33325 17.9603 5.33325 17.5C5.33325 17.0398 5.70635 16.6667 6.16659 16.6667C6.62682 16.6667 6.99992 17.0398 6.99992 17.5ZM16.1666 17.5C16.1666 17.9603 15.7935 18.3334 15.3333 18.3334C14.873 18.3334 14.4999 17.9603 14.4999 17.5C14.4999 17.0398 14.873 16.6667 15.3333 16.6667C15.7935 16.6667 16.1666 17.0398 16.1666 17.5Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
          </svg>
        </span>
        <span
          class="inline-block whitespace-nowrap overflow-visible text-current"
        >
          Add to cart
        </span>
      </button>
    </div>
  </div>
</div>
`;
